### ISSUE TITLE
Pausing/Unpausing breaks collision detection in move_and_collide

### ISSUE THREAD

**Issue by stickzman** [godotengine/godot#60577]
### Godot version

3.4.4.stable.official

### System information

Windows 10

### Issue description

I have two KinematicBody2D nodes with circle collision shapes. Body1 stays in one spot calling `move_and_collide(Vector2.ZERO)` every `_physics_process`. Body2 is moving away from Body1 every `_physics_process`.   The 2 bodies are initially overlapping when the scene starts.
If Body1's `move_and_collide` function returns a collision, I pause the SceneTree with `get_tree().set_deferred("paused", true)`, yield 1 second, then resume.

**What I expect:** A single collision is detected, the `_physics_process` concludes allowing Body2 to move away from Body1, and the game is paused for 1 second then resumes at normal speed.

**What is happening:** For every `_physics_process` after unpausing, Body1's `move_and_collide` detects the same collision with Body2, even as Body2 gets visibly farther and farther away from Body1.  This creates an infinite loop of collision detection and pausing.

### Steps to reproduce

1. Open the minimal reproduction project
2. Enable Debug > Visible Collision Shapes in the Editor
3. Run the project
4. Note the "collided!" outputs showing the collision is detected and the looped pausing, even as the collision visibly ends

Furthermore:
1. Open body1.gd
2. Comment out the line that pauses the scene tree (line 5)
3. Run the project again and notice how the scene runs correctly with that one change

### Minimal reproduction project

[paused move and collide bug example.zip](https://github.com/godotengine/godot/files/8579318/paused.move.and.collide.bug.example.zip)

**Comment by Sauermann** [comment]
Might be related to #58602.

**Comment by rburing** [comment]
If we add to the `_physics_process` function of `body2` the line

    print('body2: ', Physics2DServer.body_get_direct_state(get_rid()).transform.origin)

we see that the physics representation of `body2` doesn't move at all (so the visible collision shapes are misleading here).

This explains the repeated collisions: it is the repeated recovery of `body1` which is reported as a collision.

(The PR https://github.com/godotengine/godot/pull/59438 proposes that recovery-only should not be reported as a collision, which fixes some issues.)

I don't know why the physics representation of `body2` is not moving, though...

**Comment by stickzman** [comment]
Hi all, out of curiosity I downloaded my sample project above and ran it in 4.2.2 after converting it and the issue seems to be resolved.  No collision is reported now at all, and the collider moves away without issue.

Interestingly, if you move the bodies to be more intensely overlapping at the start, you get the same behavior of it detecting a collision, pausing, then continuing and detecting another collision, despite the fact that the colliders have visibly separated at that point, put I've only gotten it to loop this way a maximum of 3 times before returning to running normally, instead of the infinite loop I saw before.

This issue can likely be closed.  Thank you!
